scrapy/resolver.py

```diff
@@ -16,8 +16,7 @@ def __init__(self, reactor, cache_size, timeout):
     def getHostByName(self, name, timeout=None):
         if name in dnscache:
             return defer.succeed(dnscache[name])
-        if not timeout:
-            timeout = self.timeout
+        timeout = (self.timeout,)
```

Comment:
This patch indeed assumes that `DNS_TIMEOUT` is a single int or float, like [the docs mention](https://docs.scrapy.org/en/latest/topics/settings.html#dns-timeout).
Do I understand that you would like to allow users to pass a tuple of ints/floats as `DNS_TIMEOUT` setting value?